control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go

```diff
@@ -61,10 +63,21 @@ func getCloudConfig(externalNetworkID *string, cloudName string, credentialsSecr
 	if externalNetworkID != nil && *externalNetworkID != "" {
 		config += "floating-network-id = " + *externalNetworkID + "\n"
 	}
+	config += "\n[Networking]\n"
+	config += "address-sort-order = " + machineNetworksToList(machineNetwork) + "\n"
```

Comment:
@MaysaMacedo @mdbooth this is the simplest way I found to do it without having to do surgery in CPO.

Matt I remember you wanted to change CPO to find out "primary" NIC with Neutron tags. While we can investigate that, I think for now doing address-sort-order in CPO isn't a bad thing given we're not even Tech Preview and it's something we can easily (IMO) iterate later.

Reply:
Ack. As previously discussed I'm not a fan of this solution, but it's fine for now as you say.